File: src/main/java/com/github/njuro/jard/user/UserRepository.java
package com.github.njuro.jard.user;

import com.github.njuro.jard.base.BaseRepository;
import java.util.Optional;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends BaseRepository<User> {

  Optional<User> findByUsernameIgnoreCase(String username);

  User findByEmailIgnoreCase(String email);
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/user/UserRepository.kt
package com.github.njuro.jard.rewrite.user

import com.github.njuro.jard.rewrite.base.BaseRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : BaseRepository<User> {

    fun findByUsernameIgnoreCase(username: String): Optional<User>

    fun findByEmailIgnoreCase(email: String): User
}
